package/MDAnalysis/topology/RDKitParser.py

```diff
@@ -0,0 +1,289 @@
+     - Atomnames
+     - Elements
+     - Masses
+     - Bonds
+     - Resids
+     - Resnums
+     - Segids
+
+    Guesses the following:
+     - Atomtypes
+
+    Depending on RDKit's input, the following Attributes might be present:
+     - Charges
+     - Resnames
+     - AltLocs
+     - ChainIDs
+     - Occupancies
+     - Tempfactors
+     
+    """
+    format = 'RDKIT'
+
+    @staticmethod
+    def _format_hint(thing):
+        """Can this Parser read object *thing*?
+
+        .. versionadded:: 1.0.0
+        """
+        try:
+            from rdkit import Chem
+        except ImportError:  # if no rdkit, probably not rdkit
+            return False
+        else:
+            return isinstance(thing, Chem.Mol)
+
+    def parse(self, **kwargs):
+        """Parse RDKit into Topology
+
+        Returns
+        -------
+        MDAnalysis Topology object
+        """
+
+        mol = self.filename
+
+        # Atoms
+        names = []
+        resnums = []
+        resnames = []
+        elements = []
+        masses = []
+        charges = []
+        ids = []
+        atomtypes = []
+        segids = []
+        altlocs = []
+        chainids = []
+        occupancies = []
+        tempfactors = []
+
+        # check if multiple charges present
+        atom = mol.GetAtomWithIdx(0)
+        if atom.HasProp('_GasteigerCharge') and (
+        atom.HasProp('_TriposPartialCharge')
+        ):
+            warnings.warn(
+                'Both _GasteigerCharge and _TriposPartialCharge properties ',
+                'are present. Using Gasteiger charges by default.')
+
+        for atom in mol.GetAtoms():
+            ids.append(atom.GetIdx())
+            elements.append(atom.GetSymbol())
+            masses.append(atom.GetMass())
+            mi = atom.GetMonomerInfo()
+            if mi: # atom name and residue info are present
+                names.append(mi.GetName().strip())
+                resnums.append(mi.GetResidueNumber())
+                resnames.append(mi.GetResidueName())
+                segids.append(mi.GetSegmentNumber())
+                altlocs.append(mi.GetAltLoc().strip())
+                chainids.append(mi.GetChainId().strip())
+                occupancies.append(mi.GetOccupancy())
+                tempfactors.append(mi.GetTempFactor())
+            else:
+                # atom name (MOL2 only)
+                try:
+                    names.append(atom.GetProp('_TriposAtomName'))
+                except KeyError:
+                    pass
+                # atom type (MOL2 only)
+                try:
+                    atomtypes.append(atom.GetProp('_TriposAtomType'))
+                except KeyError:
+                    pass
+                # gasteiger charge (computed):
+                # if the user took the time to compute them, make it a priority
+                # over charges read from a MOL2 file
+                try:
+                    charges.append(atom.GetDoubleProp('_GasteigerCharge'))
```

Comment:
We'll want tests here to check that we are picking up the gasteiger and tripos charges as expected.